File: api/Models/Order.cs
using System.ComponentModel.DataAnnotations;

namespace BuckeyeMarketplaceApi.Models;

public enum OrderStatus
{
    Pending = 0,
    Processing = 1,
    Shipped = 2,
    Delivered = 3,
    Cancelled = 4
}

public class Order
{
    public int Id { get; set; }

    [Required]
    public string UserId { get; set; } = string.Empty;

    [Required]
    public string ConfirmationNumber { get; set; } = string.Empty;

    public DateTime OrderDate { get; set; } = DateTime.UtcNow;

    public OrderStatus Status { get; set; } = OrderStatus.Pending;

    public decimal Total { get; set; }

    [Required]
    public string ShippingAddress { get; set; } = string.Empty;

    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
}

File: frontend/src/types/Order.ts
export interface OrderItem {
  id: number;
  productId: number;
  productName: string;
  unitPrice: number;
  quantity: number;
  lineTotal: number;
}

export interface Order {
  id: number;
  confirmationNumber: string;
  orderDate: string;
  status: string;
  total: number;
  shippingAddress: string;
  userEmail?: string | null;
  items: OrderItem[];
}

export interface CreateOrderRequest {
  shippingAddress: string;
}
